src/rpc/rpc.h

```diff
@@ -92,74 +101,35 @@ struct RPCContext {
    */
   std::shared_ptr<ServerState> server_state;
 
-  /*! \brief Get the thread-local RPC context structure */
-  static RPCContext *ThreadLocal() {
-    return dmlc::ThreadLocalStore<RPCContext>::Get();
+  /*! \brief Get the RPC context singleton */
+  static RPCContext* getInstance() {
```

Comment:
do we need really singleton? if yes, we could use below one which is thread-safe since C++11.
static RPCContext& getInstance() {static RPCContext ctx; return ctx; }

Reply:
Fixed